compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/NodeFinder.java

```diff
@@ -0,0 +1,1484 @@
+        lookupNode((BLangNode) joinClause.onClause);
+    }
+
+    @Override
+    public void visit(BLangLetClause letClause) {
+        for (BLangLetVariable var : letClause.letVarDeclarations) {
+            lookupNode((BLangNode) var.definitionNode);
+        }
+    }
+
+    @Override
+    public void visit(BLangOnClause onClause) {
+        lookupNode(onClause.lhsExpr);
+        lookupNode(onClause.rhsExpr);
+    }
+
+    @Override
+    public void visit(BLangOrderKey orderKeyClause) {
+        lookupNode(orderKeyClause.expression);
+    }
+
+    @Override
+    public void visit(BLangOrderByClause orderByClause) {
+        for (OrderKeyNode key : orderByClause.orderByKeyList) {
+            lookupNode((BLangNode) key);
+        }
+    }
+
+    @Override
+    public void visit(BLangSelectClause selectClause) {
+        lookupNode(selectClause.expression);
+    }
+
+    @Override
+    public void visit(BLangWhereClause whereClause) {
+        lookupNode(whereClause.expression);
+    }
+
+    @Override
+    public void visit(BLangDoClause doClause) {
+        lookupNode(doClause.body);
+    }
+
+    @Override
+    public void visit(BLangOnConflictClause onConflictClause) {
+        lookupNode(onConflictClause.expression);
+    }
+
+    @Override
+    public void visit(BLangLimitClause limitClause) {
+        lookupNode(limitClause.expression);
+    }
+
+    @Override
+    public void visit(BLangWhile whileNode) {
+        lookupNode(whileNode.expr);
+        lookupNode(whileNode.body);
+    }
+
+    @Override
+    public void visit(BLangLock lockNode) {
+        lookupNode(lockNode.body);
+    }
+
+    @Override
+    public void visit(BLangTransaction transactionNode) {
+        lookupNode(transactionNode.transactionBody);
+    }
+
+    @Override
+    public void visit(BLangTupleDestructure stmt) {
+        lookupNode(stmt.expr);
+        lookupNode(stmt.varRef);
+    }
+
+    @Override
+    public void visit(BLangRecordDestructure stmt) {
+        lookupNode(stmt.expr);
+        lookupNode(stmt.varRef);
+    }
+
+    @Override
+    public void visit(BLangErrorDestructure stmt) {
+        lookupNode(stmt.expr);
+        lookupNode(stmt.varRef);
+    }
+
+    @Override
+    public void visit(BLangForkJoin forkJoin) {
+        lookupNodes(forkJoin.workers);
+    }
+
+    @Override
+    public void visit(BLangWorkerSend workerSendNode) {
+        if (setEnclosingNode(workerSendNode.expr, workerSendNode.expr.pos)) {
+            return;
+        }
+
+        setEnclosingNode(workerSendNode, workerSendNode.workerIdentifier.pos);
+    }
+
+    @Override
+    public void visit(BLangWorkerReceive workerReceiveNode) {
+        setEnclosingNode(workerReceiveNode, workerReceiveNode.workerIdentifier.pos);
+    }
+
+    @Override
+    public void visit(BLangRollback rollbackNode) {
+        lookupNode(rollbackNode.expr);
+    }
+
+    @Override
+    public void visit(BLangLiteral literalExpr) {
+        // Do nothing
```

Comment:
Shall we introduce a base visitor with empty implementation for all the visitor methods? We have a couple of visitors now and for some node we have to duplicate this empty visitor methods.